### ISSUE TITLE
otknl does not build

### ISSUE THREAD

**Issue by xrchz** [HOL-Theorem-Prover/HOL#660]
When I try building on current master (f535214bfea72da93f045caf3e8a2e2a485bed6f) with the `--otknl` flag, I run into this error:
```
src/opentheory/postbool/Logging.sml:740: error: Value or constructor (set_definition_callback) has not been declared in structure Thm
```

More importantly, it seems that the `--otknl` build option is never tested automatically, and hence subject to rapid bit-rot. I would like one outcome of this issue to be adding `--otknl` to the automated regression tests.

**Comment by oskarabrahamsson** [comment]
@xrchz suggested that 2cf8aa078 could be the culprit. The changes to `src/thm/Holmakefile` replace `Thm.ui` with `$(FTHM_UI)`, which gives the wrong `Thm` signature.

I can confirm that after reverting this commit, the build succeeds. I don't know what the proper fix would be, though.

**Comment by mn200** [comment]
I agree that we should have `--otknl` tested; I do have it on a cron job, but for reasons I can't puzzle out, the cron environment always fails to find the `opentheory` tool (and yes, I do set the `PATH` environment variable in the `crontab` file).

**Comment by mn200** [comment]
There is now an opentheory build tested in Travis (f4477a5f, ce2674af), and I have fixed the compilation error (c3d57abdd, ac1c33070)

**Comment by binghe** [comment]
Hi Michael, the Moscow ML build is broken again at current `master`, blocked by something called `Word64.*`. looks like a simple build order problem.
